Remind AI tools of disclosure policy direct in SECURITY.md (#9778)
geoserver/geoserver: SECURITY.md

Author: ```suggestion
or work with one of the [commercial support providers](https://geoserver.org/support/) who participate on behalf of their customers.

We advise security researchers and national agencies that linking to or attempting to enforce their own disclosure policies or timelines is unwelcome and inappropriate conduct. The recommended course of action is to work with your clients on providing or resourcing a fix.
```

2. AI assistance to locate issues is allowed. Please read the [AI policy](https://geoserver.org/ai/) and make sure to follow it.

   Be advised that GeoServer is a community project with a clear coordinated vulnerability disclosure
   policy allowing the members of our community to update prior to public disclosure.

>    Reports assuming a vendor relationship, or linking to their own disclosure policy, is not appropriate.

Author: ```suggestion
   Reports assuming a vendor relationship, or linking to their own disclosure policy, are not appropriate.
```

>    Reports assuming a vendor relationship, or linking to their own disclosure policy, are not appropriate.

Reviewer: I wonder if we should be more assertive.

```suggestion
   Reports assuming a vendor relationship, or linking to their own disclosure policy, are not appropriate.
   
   Repeat non-community language may result in reports being given lower priority. In extreme cases, reports will be closed without action. 
```

While I like the idea of security reporting, maintaining the community is a key issue.

Author: I think we should repeat, but further down the page in the "responsible disclosure" section... let me make a suggestions for your consideration @bradh

Reviewer: Lgtm

>    Reports assuming a vendor relationship, or linking to their own disclosure policy, are not appropriate.

Reviewer 2: I think this should be reworded somehow, not sure many will understand what "assuming a vendor relationship" even means. A security researcher might be focused on finding issues in the code and reporting them, without even thinking about ramifications in terms of addressing and solving the issue (a "not my problem" mentality).

Author: Okay I will remove that phrase.

Author: And focusing on finding issues is fine, setting an expectation of service, and timeframe for the community to respond is not fine.

>    Reports assuming a vendor relationship, or linking to their own disclosure policy, are not appropriate.

Author: ```suggestion
   Reports linking to their own disclosure policy are not appropriate.
```

This represents a balance between transparency and participation that does not overwhelm participants. 
Those seeking greater visibility are encouraged to volunteer with the geoserver-security list;
or work with one of the [commercial support providers](https://geoserver.org/support/) who participate on behalf of their customers.

> We advise security researchers and national agencies that linking to or attempting to enforce their own disclosure policies or timelines is unwelcome and inappropriate conduct. The recommended course of action is to work with your clients on providing or resourcing a fix.

Author: Mention collaboration as an recommendation:
```suggestion
We advise security researchers and national agencies that linking to or attempting to enforce their own disclosure policies or timelines is unwelcome and inappropriate conduct. The recommended course of action is to direct your clients to collaborating on, providing, or resource a fix.
```
